webserver.h

```diff
@@ -43,7 +44,60 @@ class WebServer
 
     void Start()
     {
-        // Define the `/api/canvases` endpoint
+        // Define the `/api/sockets` endpoint
+        CROW_ROUTE(_crowApp, "/api/sockets")
+            .methods(crow::HTTPMethod::GET)([&]() -> crow::response
+            {
+                auto socketsJson = nlohmann::json::array();
+                
+                for (size_t canvasId = 0; canvasId < _allCanvases.size(); ++canvasId) {
+                    if (!_allCanvases[canvasId]) {
+                        continue;
+                    }
+
+                    for (size_t featureId = 0; featureId < _allCanvases[canvasId]->Features().size(); ++featureId) {
+                        nlohmann::json socketJson;
+                        to_json(socketJson, _allCanvases[canvasId]->Features()[featureId]->Socket());
+                        socketJson["featureId"] = featureId;
+                        socketJson["canvasId"] = _allCanvases[canvasId]->Id();
+                        socketsJson.push_back(socketJson);
+                    }
+                }
+                
+                return socketsJson.dump(); 
+            });
+
+        // Define the `/api/sockets/:id` endpoint
+        CROW_ROUTE(_crowApp, "/api/sockets/<int>")
+            .methods(crow::HTTPMethod::GET)([&](int socketId) -> crow::response
+            {
+                // Search through all canvases and features to find the matching socket
+                for (size_t canvasId = 0; canvasId < _allCanvases.size(); ++canvasId) {
+                    if (!_allCanvases[canvasId]) {
+                        continue;
+                    }
+
+                    for (size_t featureId = 0; featureId < _allCanvases[canvasId]->Features().size(); ++featureId) {
+                        const auto& feature = _allCanvases[canvasId]->Features()[featureId];
+                        
+                        // Check if this socket matches the requested ID
+                        if (feature->Socket().Id() == socketId) {
+                            nlohmann::json socketJson;
+                            to_json(socketJson, feature->Socket());
+                            
+                            // Add the contextual fields
+                            socketJson["featureId"] = featureId;
+                            socketJson["canvasId"] = _allCanvases[canvasId]->Id();
+
+                            return socketJson.dump();
+                        }
+                    }
+                }
+                
+                // If we didn't find the socket, return a 404
+                return crow::response(404, "Socket not found"); 
```

Comment:
For the purpose of consistency with the canvas endpoint:

```suggestion
                return {crow::NOT_FOUND, R"({"error": "Socket not found"})"}; 
```